### PR TITLE
Add member lists to the /projects endpoints.

### LINKED ISSUE #239: [8] Test the project member lists
The projects model will be changed to add a "users" field, that looks like the following:

    "users": {
        "user1": {member: true, spectator: false, manager: false},
        "user2": {member: true, spectator: true, manager: true},
        ...
    }

### PR DESCRIPTION
**Pull request by LyonesGamer**
Fixes #239 and #240.

### CODE REVIEW

File: tests/projects.js [hunk 1]
```diff
@@ -32,6 +32,73 @@ module.exports = function(expect, request, baseUrl) {
...
+      uri: 'https://code.osuosl.org/projects/ganeti-webmgr',
+      name: 'Ganeti Web Manager',
+      slugs: ['ganeti-webmgr', 'gwm'],
+      deleted_at: null,
+      updated_at: null,
+      created_at: '2014-01-01',
+      uuid: 'c285963e-192b-4e99-9d92-a940519f1fbd',
+      revision: 1,
+      users: {
+        tschuy: {member: true, spectator: true, manager: true},
```
Comment (athai) [inline tests/projects.js]: Project users and their permissions don't match what's currently in test_data.json

File: tests/projects.js [hunk 2]
```diff
@@ -32,6 +32,73 @@ module.exports = function(expect, request, baseUrl) {
...
+      name: 'Whats Fresh',
+      slugs: ['wf'],
+      deleted_at: null,
+      updated_at: null,
+      created_at: '2014-01-01',
+      uuid: '9369f959-26f2-490d-8721-2948c49c3c09',
+      revision: 1,
+      users: {
+        deanj: {member: true, spectator: true, manager: false},
+        thai: {member: false, spectator: true, manager: false},
```
Comment (athai) [inline tests/projects.js]: Add tschuy to list of members

File: tests/projects.js [hunk 3]
```diff
@@ -32,6 +32,73 @@ module.exports = function(expect, request, baseUrl) {
...
+    {
+      uri: 'https://github.com/osuosl/timesync',
+      name: 'Timesync',
+      slugs: ['timesync', 'ts'],
+      deleted_at: null,
+      updated_at: null,
+      created_at: '2014-01-01',
+      uuid: '1f8788bd-0909-4397-be2c-79047f90c575',
+      revision: 1,
+      users: {},
```
Comment (athai) [inline tests/projects.js]: Add yourself to the list of members

File: tests/projects.js [hunk 4]
```diff
@@ -32,6 +32,73 @@ module.exports = function(expect, request, baseUrl) {
...
+    {
+      uri: 'https://github.com/osuosl/chiliproject',
+      name: 'Chili Project',
+      slugs: [],
+      deleted_at: '2014-01-01',
+      updated_at: null,
+      created_at: '2009-07-07',
+      uuid: '6abe7f9a-2c4b-4c1d-b4f9-1222b47b8a29',
+      revision: 1,
+      users: {},
```
Comment (athai) [inline tests/projects.js]: https://github.com/osuosl/timesync-node/blob/tristan/members-list/tests/fixtures/test_data.json#L434

**Comment by ElijahCaine** [comment]
I didn't go through this with a fine-toothed comb, but it looks good to me. We will find the small problems when we implement. +1

File: tests/projects.js [hunk 5]
```diff
@@ -504,6 +444,134 @@ module.exports = function(expect, request, baseUrl) {
...
+
+          // expect body of post request to be the new state of gwm
+          expect(body).to.deep.equal(expectedPost);
+
+          checkListEndpoint(done, expectedResults, token);
+        });
+      });
+    });
+
+    it('successfully adds demotes an existing user', function(done) {
```
Comment (athai) [inline tests/projects.js]: Remove the extra word

File: tests/projects.js [hunk 6]
```diff
@@ -504,6 +444,134 @@ module.exports = function(expect, request, baseUrl) {
...
+
+          // expect body of post request to be the new state of gwm
+          expect(body).to.deep.equal(expectedPost);
+
+          checkListEndpoint(done, expectedResults, token);
+        });
+      });
+    });
+
+    it('successfully adds removes the calling user', function(done) {
```
Comment (athai) [inline tests/projects.js]: Here too

File: tests/projects.js [hunk 7]
```diff
@@ -1573,34 +1645,46 @@ module.exports = function(expect, request, baseUrl) {
...
+      uri: 'https://code.osuosl.org/projects/ganeti-webmgr',
+      name: 'GANETI WEB MANAGER',
+      uuid: 'c285963e-192b-4e99-9d92-a940519f1fbd',
+      revision: 2,
+      deleted_at: null,
+      updated_at: currentTime,
+      created_at: '2014-01-01',
+      slugs: ['ganeti-webmgr', 'gwm'],
+      users: {
+        patcht: {member: true, spectator: true, manager: true},
```
Comment (athai) [inline tests/projects.js]: Wrong users are listed under GWM?

**Comment by athai** [comment]
+1

**Comment by Kennric** [comment]
Tests pass and I don't see any issues with the logic, but definitely get another review or two. +1

**Comment by ElijahCaine** [comment]
+1 let's merge

**Pull request merged**